### ISSUE TITLE
mnist_hogwild - breaking gradient sharing

### ISSUE THREAD

**Issue by pfre00** [pytorch/examples#138]
The [loop](https://github.com/pytorch/examples/blob/28471f2132aede964ec3674d30d6e551f829a363/mnist_hogwild/train.py#L10) which is supposed to be breaking gradient sharing in mnist_hogwild doesn't seem to be doing anything. ```param.grad is not None``` evaluates to false, since ```param.grad``` is allocated lazily, in the subprocesses. There, every process allocates gradient tensors separately (i think?), so there might be no need for breaking gradient sharing manually at all.

**Comment by soumith** [comment]
you are correct. in v0.1.9 we introduced allocating gradients lazily, and hence you see this.

**Comment by apaszke** [comment]
Yes, there's no need anymore and the loop should be removed.
